<a href="https://colab.research.google.com/github/Danicoder/Especializacion_IA_BigData/blob/main/Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
##Subiendo el Archivo
from google.colab import files
load=files.upload()

Saving titanic_train.csv to titanic_train.csv


In [19]:
import pandas as pd

##LECTURA DEL ARCHIVO
df = pd.read_csv('titanic_train.csv')

#Imprimir 20 primeras filas
df.head(20)

#Imprimir las últimas 5 filas
df.tail(5)

#Imprime la información del fichero para saber el tipo de dato
df.info()

#Imprime el número de filas y columnas
df.shape

#Imprime resumen estadístico
df.describe()

titanic= pd.read_csv("titanic_train.csv")
#Imprimir datos con pasajeros > 30 años
print("\n =====Pasajeros > 30===== \n")
pasajeros_mayor_30 =titanic[titanic['Age']>30]
print(pasajeros_mayor_30.describe())

#Imprimir pasajeros con edad comprendida entre 45 y 55 años
pasajeros_entre_45_55= titanic[(titanic['Age']>=45) & (titanic['Age']<=55)]
print("\n =====Datos de pasajeros > 45 y < 55===== \n")
print(pasajeros_entre_45_55.describe())

#Para saber la cantidad de pasajero que cumplen la condición
print("\n =====cantidad de pasajero de entre 45 y 55 años: ===== \n")
cantidad = pasajeros_entre_45_55['PassengerId'].count()
print(cantidad)

#Pasajeros que viajan solos
pasajeros_singles= titanic[titanic['SibSp']==0]
print("\n =====cantidad de pasajero viajan solos===== \n")
print(pasajeros_singles.shape[0]) #shape[0] --> solo devuelve nº filas

#Filas de pasajeros que sobrevivieron
sobreviviventes = titanic.loc[titanic["Survived"] == 1]
print("\n =====Sobrevivientes=====")
print(sobreviviventes.shape[0])

#filas de hombres que sobrevivieron
sobreviviventes_h = titanic.loc[(titanic["Survived"] == 1) & (titanic["Sex"] == "male")]
print("\n =====Hombres sobrevivientes=====")
print(sobreviviventes_h.shape[0])

#Número de columnas y filas
print("\n =====Edad y Sexo de los pasajeros=====")
print(titanic[["Age","Sex"]].shape)

#Pasajeros con cabina 2 y 3
pasajeros_cabina_2_3 = titanic[(titanic['Pclass']==2) | (titanic['Pclass']==3)]
print("\n =====Pasajeros con cabina 2 y 3=====")
print(pasajeros_cabina_2_3.head())

#Pasajeros con edad != 0
pasajeros_edad_notnull =titanic[titanic['Age'].notna()]
print("\n =====Pasajeros con edad != 0=====")
print(pasajeros_edad_notnull.describe())
pasajeros_edad_notnull.shape

#Pasajeros entre las filas 15 - 30 y columnas 4-7
pasajeros_filas_col= titanic.iloc[14:30,3:7]
print("\n =====Pasajeros entre las filas 15 - 30 y columnas 4-7=====")
print(pasajeros_filas_col)

#Filas compredidas entre 15 - 30 y columnas entre 1-4
pasajeros_filas_col2= titanic.iloc[14:30,0:4]
print("\n =====Filas compredidas entre 15 - 30 y columnas entre 1-4=====")
print(pasajeros_filas_col2)

#Edad pasajeros Titanic (20 primeros)
edad = titanic["Age"]
print("\n =====Edad pasajeros Titanic=====")
print(edad.head(20))

#Calcula la edad media de los pasajeros del Titanic
edad_media=titanic["Age"].mean()
print("\n =====Edad media de los pasajeros del Titanic=====")
print(edad_media)
#Valor medio de edad y precio del billete de los pasajeros
print("\n =====Valor medio de edad y precio del billete de los pasajeros=====")
media_edad = titanic["Age"].mean()
precio = titanic["Fare"]*titanic['Parch']
print(media_edad)
print(precio)

#valor medio, promedio mínimo, máximo, percentiles de la Edad y el precio del billete

print("\n =====Valor medio,promedio min/max,percentil de la edad y precio del billete=====")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB

 =====Pasajeros > 30===== 

       PassengerId    Survived      Pclass         Age       SibSp  \
count   305.000000  305.000000  305.000000  305.000000  305.000000   
mean    467.659016    0.406557    1.927869   42.927869    0.347541   
std  

1.   ¿Cuántos pasajeros mayores de 30 años te salen? 305
2.   Como sacarías los 15 primeros de la lista anterior: df.tail(15)
3.   ¿Cuántos pasajeros tienen entre 45 y 55 años? 35.660 pasajeros
4.   ¿Cuántos pasajeros viajan solos? 276.865
5.   ¿Cúantos filas de pasajeros sobrevivieron? 342
6.   ¿Cúantos filas de pasajeros hombres sobrevivivieron? 109
7.   Elemento de lista



Total pasajeron: 397.386







